### PR TITLE
[12.0][Add] partner_company_group

### PR DESCRIPTION
**Pull request by tschanzt**
Adds a company group field on the partner and a related field on lead, sale order and invoice to allow grouping and searching for the company group.

### CODE REVIEW

File: partner_company_group/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    'category': 'Sales',
+    'author': 'Camptocamp',
+    'license': 'AGPL-3',
+    'depends': [
+        'base',
+        'account',
+        'crm',
+        'sale',
+    ],
+    'website': 'http://www.camptocamp.com',
```
Comment (nikul-serpentcs) [inline partner_company_group/__manifest__.py]: `https://github.com/OCA/partner-contact`

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
netpicking

**Comment by tschanzt** [comment]
@nikul-serpentcs updated.

File: partner_company_group/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    company_group_id = fields.Many2one(
+        related='partner_id.company_group_id',
```
Comment (mpanarin) [inline partner_company_group/models/account_invoice.py:11]: readonly?

File: partner_company_group/models/crm_lead.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
+
+
+class CrmLead(models.Model):
+    _inherit = 'crm.lead'
+
+    company_group_id = fields.Many2one(
+        related='partner_id.company_group_id',
```
Comment (mpanarin) [inline partner_company_group/models/crm_lead.py:11]: same

File: partner_company_group/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    company_group_id = fields.Many2one(
+        related='partner_id.company_group_id',
```
Comment (mpanarin) [inline partner_company_group/models/sale_order.py:11]: same

**Review by mpanarin — CHANGES_REQUESTED** [review]

File: partner_company_group/models/account_invoice.py (hunk 2 above)
Comment (nikul-serpentcs) [reply partner_company_group/models/account_invoice.py:11]: @mpanarin No need to `readonly = True` when a field is related.

Everywhere

Comment (mpanarin) [reply partner_company_group/models/account_invoice.py:11]: You will be able to change the field and, therefore, changing the value on a different record completely. This is not very user-friendly and obscure, as well as should be avoided. 
So I still think they should be readonly

Comment (tschanzt) [reply partner_company_group/models/account_invoice.py:11]: @mpanarin related fields are readonly=true as a default in v12. although i would have forgotten it.
https://github.com/odoo/odoo/blob/12.0/odoo/fields.py#L427

Comment (nikul-serpentcs) [reply partner_company_group/models/account_invoice.py:11]: @mpanarin In Odoo Base version 12, related always `readonly = True` look like a `compute/ Functional` fields, so no need to added this type of attribute

**Review by simahawk — APPROVED** [review]

**Review by leemannd — APPROVED** [review]

**Review by leemannd — APPROVED** [review]

**Review by mpanarin — APPROVED** [review]

**Review by gurneyalex — APPROVED** [review]

**Pull request merged**